scalding-beam/src/test/scala/com/twitter/scalding/beam_backend/BeamBackendTests.scala

```diff
@@ -389,6 +389,31 @@ class BeamBackendTests extends FunSuite with BeforeAndAfter {
     )
   }
 
+  test("Force to Disk execution") {
+    val bmode = BeamMode.default(pipelineOptions)
+    val tmpPath1 = tmpPath("tp1")
+    val tmpPath2 = tmpPath("tp2")
+
+    val forcedExecution =
+      TypedPipe
+        .from(Seq(5, 3, 2, 6, 1, 4))
+        .forceToDiskExecution
```

Comment:
Tested by adding println() statements and they got printed only once. Whats a good way to add a test which checks that logic is not computed twice?
I tried by defining AtomicLong() and incrementing it in a map function but that didn't work

Reply:
this is a really good question.

I think atomic long isn't going to work because that assumes that there is no serialization of the job, which I think there likely is.

What might work, but will be ugly, would be to make a `TypedPipe.from(List(1)).map { x => /* create a specific file, or if it already exists fail */ }`

So, allocate a single random path at the start of the test, make sure the map function creates that file, then at the end of the test remove the path.

That would be my suggestion:

https://docs.oracle.com/javase/6/docs/api/java/io/File.html#createNewFile()